doc/source/how-to-use-clientapp.rst

```diff
@@ -0,0 +1,303 @@
+
+As you can see, we can easily reuse existing Flower clients with the :code:`ClientApp` by adding 3 lines of code! Let’s now walk through how to use the low-level API for greater implementation flexibility in our projects.
+
+Using low-level API
+-------------------
+
+With Flower 1.8, we provide a set of low-level APIs to allow more versatile ways to implement any functionality that we like in the :code:`ClientApp`. The three functions that can be registered in a :code:`ClientApp` are:
+
+* :code:`app.train()`
+* :code:`app.evaluate()`
+
+which runs the training and evaluation on client-side data. 
+
+.. admonition:: Important
+    :class: important
+
+    As we continuously enhance Flower at a rapid pace, we'll periodically update the functionality and this how-to document. Please feel free to share any feedback with us!
+
+Next, we have a key part in the low-level API, which is the message passing format. In Flower 1.8, we introduced |message_link|_ and |context_link|_ objects. These are abstractions that unify the message formats that are relayed back and forth in a Flower project. When compared to the high-level example above, objects such as :code:`config` and :code:`parameter` need to be separately configured and tracked in each method (such as :code:`get_parameters()`, :code:`set_parameters()`, :code:`fit()`, and :code:`evaluate()`) which can make the workflow un-intuitive. To simplify the usage, we unify the data format via the :code:`Message` and :code:`Context` abstractions.
+
+A :code:`Message` contains different information depending on whether it is received or returned by the :code:`ClientApp`: A received :code:`Message` contains information required to run the tasks within each function, such as training configurations and parameters. A returned :code:`Message` on the other hand, contains results of the computation, such as loss or accuracy metrics. While a :code:`Message` wraps information that is consumed within a round, a :code:`Context` object wraps information that is persisted across each round of federated learning. One example where :code:`Context` can be used is to track the metrics from an earlier round (or rounds), which is handy, such as when early stopping strategies are implemented.
+
+.. admonition:: Tip
+    :class: note
+
+    The Message and Context objects contain data such as parameters, metrics, configs, and identifiers of the current run. All of these information is used by the :code:`ClientApp` to determine what and how to run the task.
+
+In short:
+
+#. :code:`Message` is a simplified messaging pattern between :code:`ClientApp` and :code:`ServerApp`, and is consumed within a round.
+#. :code:`Context` persists messages across rounds.
```

Comment:
`Context` is a `RecordSet` object, which is the same type of object carrying the bulk of the information inside a `Message` (i.e. `Message.content` is of type `RecordSet`) -- not sure if this level of detail is needed here but I feel it's better not to define `Context` with the word "message".

Reply:
Good point, I've also associated `Context` with information in the sentence above. Have revised the wordings.